cnd/src/bitcoin.rs

```diff
@@ -13,6 +13,50 @@ use serde::{
 };
 use std::{fmt, str::FromStr};
 
+/// The available Bitcoin networks.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Network {
```

Comment:
Well, since I was doing something totally new I thought I'd experiment with this - the three networks has been bothering me since we added them and no number of attempts at adding the Ethereum side has changed that (I've made 2 separate concerted efforts at it).

Reply:
Fair, maybe that kind of use of the type system is too much :)
Do you think we should turn around on this topic and go back to enums?

Reply:
Yes, I do not think these three 'network' structs are a good use of the language features.  Also when we remove rfc003 I believe there original usecase is going to disappear (if I understood correctly why we added them in the first place). This PR shows that an enum serves us just nicely.

Reply:
Why is this introduced now and was not raised earlier? 

I'm afraid that introducing this here now means for sure that we have to refactor one or the other in the future. 
I'm not against this but we should stick to one solution and not jump back and forth.

Reply:
> I'm afraid that introducing this here now means for sure that we have to refactor one or the other in the future.

We already have to do that because `Ethereum` is still using an enum :D
I am fine with moving back on this one, was a bad idea.